rfcs/2020-09-28-cloudwatch-metrics.md

```diff
@@ -0,0 +1,181 @@
+# RFC #3643 - 2020-09-28 CloudWatch Metrics source RFC
+
+This RFC proposes an implementation for a new metrics source to ingest metrics
+from AWS CloudWatch. The proposed implementation is fairly similar to the one
+used by telegraf which scrapes `GetMetricData` on a regular interval.
+
+It is probably worth reviewing [Amazon CloudWatch
+Concepts](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html)
+before looking at this if you you are unfamiliar with CloudWatch.
+
+## Scope
+
+This RFC will simply cover a new `aws_cloudwatch_metrics` source.
+
+## Motivation
+
+Users want to collect and forward metrics from AWS CloudWatch to monitor
+infrastructure and deployed services in AWS.
+
+## Internal Proposal
+
+We will add a `aws_cloudwatch_metrics` source that scrapes the AWS API
+([`GetMetricData`](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html))
+on a user-defined interval to collect metrics and forward them through the
+user's pipeline.
+
+## Doc-level Proposal
+
+The configuration of the new `aws_cloudwatch_metrics` source will look like:
+
+```toml
+[sources.cloudwatch]
+  type = "aws_cloudwatch_metrics"
+  assume_role = "arn:aws:iam::123456789098:role/my_role" # optional, no default
+  endpoints = ["127.0.0.0:5000/path/to/service"] # optional, no default, relevant when regions = []
```

Comment:
I wasn't sure what "endpoints" meant in this situation, found this:

https://docs.aws.amazon.com/general/latest/gr/cw_region.html